-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA256

- From daa686d0498d484166bf7582f2be3155b712c8e0 Mon Sep 17 00:00:00 2001
From: Sebastian Bach <sebtomba@gmail.com>
Date: Tue, 20 Jun 2023 14:18:51 +0200
Subject: [PATCH] [PM-6003] Improve the calculation of MN transaction weight

- ---
 pallets/midnight/src/weights.rs | 9 ++-------
 1 file changed, 2 insertions(+), 7 deletions(-)

diff --git a/pallets/midnight/src/weights.rs b/pallets/midnight/src/weights.rs
index c4e90661..fbd01c06 100644
- --- a/pallets/midnight/src/weights.rs
+++ b/pallets/midnight/src/weights.rs
@@ -1,14 +1,11 @@
- -use frame_support::{
- -	traits::Get,
- -	weights::{constants, Weight},
- -};
+use frame_support::weights::Weight;
 
 use std::marker::PhantomData;
 
 pub const FIXED_MN_TRANSACTION_WEIGHT: u64 = 1_000_000_000;
 pub const FIXED_MN_TRANSACTION_COUNT: u64 = 10;
 pub const FIXED_MN_BLOCK_WEIGHT: Weight =
- -	Weight::from_parts((FIXED_MN_TRANSACTION_COUNT + 1u64) * FIXED_MN_TRANSACTION_WEIGHT, 1_000u64);
+	Weight::from_parts(FIXED_MN_TRANSACTION_COUNT * FIXED_MN_TRANSACTION_WEIGHT, 1_000u64);
 
 pub trait WeightInfo {
 	fn mn_transaction_weight(midnight_tx: &Vec<u8>) -> Weight;
@@ -18,13 +15,11 @@ pub struct MidnightWeight<T>(PhantomData<T>);
 impl<T: frame_system::Config> WeightInfo for MidnightWeight<T> {
 	fn mn_transaction_weight(_midnight_tx: &Vec<u8>) -> Weight {
 		Weight::from_parts(FIXED_MN_TRANSACTION_WEIGHT, 0)
- -			.saturating_add(T::DbWeight::get().writes(1 as u64))
 	}
 }
 
 impl WeightInfo for () {
 	fn mn_transaction_weight(_midnight_tx: &Vec<u8>) -> Weight {
 		Weight::from_parts(FIXED_MN_TRANSACTION_WEIGHT, 0)
- -			.saturating_add(constants::RocksDbWeight::get().writes(1 as u64))
 	}
 }
- -- 
2.46.0

-----BEGIN PGP SIGNATURE-----

iQIzBAEBCAAdFiEE9Oz16eF+Nv5C7TpSdEmbFI1hGmwFAmhtArIACgkQdEmbFI1h
GmxLxQ/+OzUn22ZlFhzWnPsoEu3F1MBrPAv3CjIwJ0fQe+nKZ4KqcCj8K6TPWntG
5x266Rgzx6iGMGOFgBqg5YzaxhV/1sHnrHuskInLMDh/wuoW9HRNCzcxx2Omy2WN
2JS1n/KDcQuYQaus9o0XQtfuV5J2wwxIJedZ1aKG8mAoXLJwdoFkx6VzAdDNToTd
tyHmfR7ffUMmyHCynMnjVa6juJZbLtjqFbBjii+yoDcROB7zdUvhxwMFeZ5JUYk+
zlHrVcrfCyb+sEorFSnkY+m8w9QpGYinOZnxuyITpkxPaCrgivqjifV5nUiRisWR
LHIO4tg/xZCWznNdkK1Yet5qsqlqqnVLupVrLVSNhh6lG/V/6g3d/xwoDwZHUG9V
NCUehgVyZ/qTq7e0UvHRR1mwzaEmgcI/UBDm/dxJQWGsAv1yMzpfXBqSIIkBJoVX
TTS6lClk+CHHs+P9OcUBVkUwdaPoQEcqRsju5Wd90jXqVsHPzQmEUCkQGGdFjcF4
HJrshF3L0xp7oz415bwyBeKbFAfSTdFa0Tck0lVlnm3xaBIYTdWtRYgwXK2aQpnJ
U0zht5FmcXbUUhJ7bahFkBiTHuuYr7Pkilk84f5uqQp1mnrasjY7aXat+2sw1wzP
/+HxUgHEo1ekfATgzixBk5m/DZhuwfO43tmWIXtv+Rlumx6YtY8=
=mmjT
-----END PGP SIGNATURE-----
